src/twisted/internet/defer.py

```diff
@@ -159,6 +165,65 @@ def maybeDeferred(f, *args, **kw):
         return succeed(result)
 
 
+def unbox(deferred, raises=True, passthrough=True):
```

Comment:
So I think the behavior I expected before reading this (even though I'm no deferred expert) is to always raise, and always passthrough.

Someone who wanted `raise=False` can try/except and someone who wanted specifically the failure and not the exception sounds like they'll be unhappy but I think that's the "fault" of the interaction between failures and exceptions, not of your code here -- i.e., if that was a common use case, I'd expect to see an inverted failure object which is an exception that carries around the original failure.

As for passthrough which it sounds like is more important to address -- I'd have expected always passthrough for a bunch of reasons, some dogmatic and some practical

0) I'm discarding the asymmetric options immediately because I'd find it surprising to treat failures differently than successes personally
1) no side effects always being better than side effects
2) the typical use case for me is like having a webapp which passes deferreds all around, and then I have some way of synchronizing those, but other internals there like "log deferred errors" still might be `addErrback`ing the deferred I get my hands on, and I'd still want those to work